CMake/SlicerBlockPlatformCheck.cmake

```diff
@@ -34,42 +34,26 @@ endif()
 
 if(Slicer_PLATFORM_CHECK)
   # See CMake/Modules/Platform/Darwin.cmake)
-  #   6.x == Mac OSX 10.2 (Jaguar)
-  #   7.x == Mac OSX 10.3 (Panther)
-  #   8.x == Mac OSX 10.4 (Tiger)
-  #   9.x == Mac OSX 10.5 (Leopard)
-  #  10.x == Mac OSX 10.6 (Snow Leopard)
-  #  11.x == Mac OSX 10.7 (Lion)
-  #  12.x == Mac OSX 10.8 (Mountain Lion)
-  if(DARWIN_MAJOR_VERSION LESS "9")
-    message(FATAL_ERROR "Only Mac OSX >= 10.5 is supported !")
+  # and https://en.wikipedia.org/wiki/Darwin_(operating_system)#Release_history
+  #  17.x == macOS 10.13 (High Sierra)
+  #  18.x == macOS 10.14 (Mojave)
+  #  19.x == macOS 10.15 (Catalina)
+  #  20.x == macOS 11 (Big Sur)
+  #  21.x == macOS 12 (Monterey)
+  if(DARWIN_MAJOR_VERSION LESS "17")
```

Comment:
```suggestion
  if(NOT DARWIN_MAJOR_VERSION GREATER_EQUAL "17")
```

Reply:
Make use of new expressions supported by the `if` command. See https://cmake.org/cmake/help/v3.13/command/if.html

The test now directly match the message reported to the user.